### PR TITLE
Add retry logic for setup of dotnet cli

### PR DESCRIPTION
**Pull request by dseefeld**
Copy init-tools.sh retry logic from corefx.  Also adding display_error_message() function.

Fixes dotnet/core-eng#1966

### CODE REVIEW

File: init-tools.sh [hunk 1]
```diff
@@ -37,6 +37,33 @@ fi
 
 echo "Running: $__scriptpath/init-tools.sh" > $__init_tools_log
 
+
```
Comment (dagood) [inline init-tools.sh]: \<nit> Extra newline

**Comment by dseefeld** [comment]
@dotnet-bot Test UWP CoreCLR x64 Debug Build please

**Comment by weshaggard** [comment]
The UWP leg is currently busted, so ignore it for this PR.

Is this an example copy of the init-tools.sh in master?

**Comment by dseefeld** [comment]
It's not an exact copy of master.  I am moving over execute_with_retry() and display_error_message()

**Comment by weshaggard** [comment]
Any reason it cannot exactly match? That would be my preference unless there is some reason it cannot.

**Comment by dseefeld** [comment]
Updated to match master.

**Comment by dseefeld** [comment]
@dotnet-bot test Linux x64 Release Build please

**Pull request merged**